File: OnlineExam/ClientApp/src/app/models/exam-models/exam.ts
import {BaseDto} from "../responses/baseDto";
import {Subject} from "../subject-models/subject";
import {Question} from "../question-models/question";
import {Classroom} from "../classroom-models/classroom";
import {Account, StudentWithScore} from "../account-models/account";
import {ScoreExam} from "../score-models/scoreExam";

export class Exam extends BaseDto{
  public name?:string;
  public start?:Date;
  public time?:number;
  public subject?:Subject;
  public countQuestion?:number;
  public countStudents?:number;
  public countScores?:number;
}

export class ExamDetail extends BaseDto{
  public name?:string;
  public start?:Date;
  public time?:number;
  public subject?:Subject;
  public questions:Question[] = [];
  public students:Account[] = [];
  public scores:ScoreExam[] = [];
}

export class ExamScore extends BaseDto{
  public name?:string;
  public start?:Date;
  public time?:number;
  public subject?:Subject;

  public countScore?:number;
  public countQuestion?:number;
  public studentWithScores:StudentWithScore[] = [];
}

File: OnlineExam/Models/Domain/Exam.cs
using OnlineExam.Common.Domain;

namespace OnlineExam.Models.Domain;

public class Exam : BaseDomain
{
    public virtual string Name { get; set; }
    public virtual DateTime Start { get; set; }
    public virtual int Time { get; set; }
    
    public virtual Subject Subject { get; set; }
    public virtual IList<Question> Questions { get; set; }
    public virtual IList<Student> Students { get; set; }
    public virtual IList<Score> Scores { get; set; }
}